tests/examples/test_01_hello_world.py

```diff
@@ -0,0 +1,60 @@
+"""Test for examples/example_01_hello_world.py using the suggested testing patterns."""
+
+from pydantic import SecretStr
+
+from examples.ex_01_hello_world import example
+from openhands.sdk import LLM, AgentBase
+from openhands.sdk.conversation import ConversationCallbackType, ConversationState
+from openhands.sdk.conversation.state import AgentExecutionStatus
+
+
+class FakeAgent(AgentBase):
+    """Fake agent that provides predictable responses for testing."""
+
+    model_config = {"frozen": False}  # Allow mutation for testing
+    step_count: int = 0
+    user_messages: list[str] = []
+    requested_write: bool = False
+    requested_delete: bool = False
+
+    def init_state(self, state: "ConversationState",
+                   on_event: "ConversationCallbackType") -> None:
+        pass
+
+    def step(
```

Comment:
Here we're just recording that we got messages that request certain actions, but it would be possibly to fire off actual tools.